src/Listener/TerminateListener.php

```diff
@@ -165,4 +179,58 @@ private function validateInputMap(array $inputMap, string $commandClass): void
             Assert::allString($value, 'Malformed input mapper for ' . $commandClass);
         }
     }
+
+    /**
+     * @psalm-return non-empty-string
+     */
+    private function getVendorDirectory(): string
+    {
+        $basePath     = realpath(getcwd());
+        $composerJson = file_get_contents($basePath . '/composer.json');
+        Assert::string($composerJson);
+
+        $composer = json_decode($composerJson, true);
+        Assert::isMap($composer);
+
+        $vendorDir = $composer['config']['vendor-dir'] ?? $basePath . '/vendor';
```

Comment:
I cannot see any test for custom vendor-dir, so I guess you have checked it manually?

I had an impression first that it might not working first, as it can be relative, but as a fallback we are using fullpath, but noticed then couple lines below realpath on the result, so I believe it's fine :)

Reply:
Yeah, had to test it manually, since we resolve the `composer.json` based on the current working directory. I considered a few options for testing, but they all required a lot of bizarre hoops and coding remnants that would have only existed for testing.

When I manually tested, I did so by changing the value of `config.vendor-dir` in an existing project, and having it install my branch as the 0.1.6 release, and then checking how chained commands worked. Prior to the changes in this branch, I got the warning about using 3rd party code; after, the notices were gone.

Reply:
Ended up using reflection to test the method, as the change to resolve home paths meant I needed to test those code paths as well.